flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -759,6 +778,24 @@ class TaskManager(val connectionInfo: InstanceConnectionInfo,
 
     task.unregisterMemoryManager(memoryManager)
   }
+
+  /**
+   * Returns CPU Load if getProcessCpuLoad method is present
+   * in the implementation of OperatingSystemMXBean
+   * else returns -1
+   *
+   * @param obj
+   * @return
+   */
+  private def fetchCPULoad(obj: Any): Any = {
+    val methodsList = classOf[com.sun.management.OperatingSystemMXBean].getMethods()
```

Comment:
This reflection call is made on every heartbeat report (all 5 secs by default).
it is actually only required to get the method once and invoke it each time we want to get the CPU load.